Refactor EditingStyle::removeStyleFromRulesAndContext into 3 parts

vitorroriz opened the pull request:
#### 5e59d43836f1c2ce4595b46c66864f52bdcfea9a
<pre>
Refactor EditingStyle::removeStyleFromRulesAndContext into 3 parts
<a href="https://bugs.webkit.org/show_bug.cgi?id=297016">https://bugs.webkit.org/show_bug.cgi?id=297016</a>
<a href="https://rdar.apple.com/157694930">rdar://157694930</a>

Reviewed by Ryosuke Niwa.

This function is too long and can be split into 3 steps.
These steps are currently commented, so let&apos;s formalize
the distinction made by comments by splitting the
function into 3 sub-functions.

* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::removeInlineStyleRedundantDueToMatchedRules):
(WebCore::EditingStyle::removeStyleInContextNotOverridenByMatchedRules):
(WebCore::EditingStyle::removeDisplayPropertyFromSpanStyleIfRedundant):
(WebCore::EditingStyle::removeStyleFromRulesAndContext):
* Source/WebCore/editing/EditingStyle.h:

Canonical link: <a href="https://commits.webkit.org/298314@main">https://commits.webkit.org/298314@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/947ff53bb135018462f8b01352a8568568543322

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/115111 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34822 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25308 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121226 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/65746 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/fff2d2a6-82c3-4069-a746-1aad4eb2841d) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/117000 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35471 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43386 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/87473 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/42258 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/a2027279-4820-4754-baef-e1f94003cfdd) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/118059 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28274 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103345 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67870 "Passed tests") | | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/27452 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21465 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64878 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97658 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21582 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124413 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/42077 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31472 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96268 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42448 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99535 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/96054 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/24439 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41258 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/19103 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/38065 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41953 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47491 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41489 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44809 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43222 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

rniwa commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1512,26 +1511,52 @@ void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node*
+    if (!m_mutableStyle)
+        return;
+
+    // 1. Handle matching rules redundancy
```
I don't think these comments are necessary now that they're split into its own function.

rniwa commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1512,26 +1511,52 @@ void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node*
+}
 
-    // 3. If this element is a span and has display: inline or float: none, remove them unless they are overridden by rules.
+void EditingStyle::removeDisplayPropertyFromSpanStyleIfNeeded(StyledElement& element)
```
I'd call it `removeDisplayPropertyFromSpanStyleIfRedundant` instead.

rniwa commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1465,17 +1465,18 @@ static void removePropertiesInStyle(MutableStyleProperties& styleToRemovePropert
-    // 2. Remove style present in context and not overridden by matched rules.
+void EditingStyle::removeStyleInContextNotOverridenByMatchedRules(StyledElement&, Node* context, MutableStyleProperties& styleFromMatchedRules)
+{
+    // Remove style present in context and not overridden by matched rules.
```
I don't think this comment is necessary given the function name repeats it.

rniwa commented on Source/WebCore/editing/EditingStyle.cpp:
```diff
@@ -1465,17 +1465,18 @@ static void removePropertiesInStyle(MutableStyleProperties& styleToRemovePropert
 }
 
-void EditingStyle::removeStyleFromRulesAndContext(StyledElement& element, Node* context)
+Ref<MutableStyleProperties> EditingStyle::removeStyleFromMatchedRules(StyledElement& element)
```
Maybe rename this to `removeInlineStyleRedundantDueToMatchedRules`?
Then we can probably just remove the comment below.

rniwa reviewed (approved):
r=me but I don't think we need to keep the comments in the code. They're largely redundant now that each function's name repeats the comment.

vitorroriz committed:
Refactor EditingStyle::removeStyleFromRulesAndContext into 3 parts
https://bugs.webkit.org/show_bug.cgi?id=297016
rdar://157694930

Reviewed by Ryosuke Niwa.

This function is too long and can be split into 3 steps.
These steps are currently commented, so let's formalize
the distinction made by comments by splitting the
function into 3 sub-functions.

* Source/WebCore/editing/EditingStyle.cpp:
(WebCore::EditingStyle::removeInlineStyleRedundantDueToMatchedRules):
(WebCore::EditingStyle::removeStyleInContextNotOverridenByMatchedRules):
(WebCore::EditingStyle::removeDisplayPropertyFromSpanStyleIfRedundant):
(WebCore::EditingStyle::removeStyleFromRulesAndContext):
* Source/WebCore/editing/EditingStyle.h:

Canonical link: https://commits.webkit.org/298314@main
